src/cli/api/local_api.py

```diff
@@ -382,20 +526,266 @@ def get_results(self, path: Path, all_results: bool) -> List[ResultType]:
+        network_nodes = self._get_network_nodes(experiment_network_slug)
+        if len(experiment_nodes) > len(network_nodes):
+            error_dict['warning'].append(f"In file {experiment_file_path}: too many nodes used in network "
+                                         f"'{experiment_network_slug}'. Maximum amount of nodes that can be used: "
+                                         f"{len(network_nodes)}")
+
+        # Check if all nodes exist and belong to this network
+        invalid_nodes = []
+        for node in experiment_nodes:
+            if node['slug'] not in network_nodes:
+                invalid_nodes.append(node['slug'])
+        if invalid_nodes:
+            if len(invalid_nodes) > 1:
+                error_dict['warning'].append(
+                    f"In file {experiment_file_path}: {invalid_nodes} do not exist or do not belong to"
+                    f" the network '{experiment_network_slug}'")
+            elif len(invalid_nodes) == 1:
+                error_dict['warning'].append(f"In file {experiment_file_path}: {invalid_nodes} does not exist or does "
+                                             f"not belong to the network '{experiment_network_slug}'")
+
+    def validate_experiment_channels(self, experiment_file_path: Path, experiment_data: Dict[str, Any], error_dict:
+                                     ErrorDictType) -> None:
+        """
+        Validate if the amount of channels (defined in the experiment.json file) are valid for 'network_slug' and if
+        all the channels exist and belong to this network.
+
+        Args:
+            experiment_file_path: The location of the experiment.json file
+            experiment_data: contents of the experiment.json file
+            error_dict: Dictionary containing error and warning messages of the validations that failed
+        """
+
+        # Check if the amount of channels are valid for this network
+        experiment_network_slug = experiment_data['asset']['network']['slug']
+        experiment_channels = experiment_data['asset']['network']['channels']
+        network_channels = self._get_channels_for_network(experiment_network_slug)
+
+        if len(experiment_channels) > len(network_channels):
+            error_dict['warning'].append(f"In file {experiment_file_path}: too many channels used in network "
+                                         f"'{experiment_network_slug}'. Maximum amount of channels that can be used: "
+                                         f"{len(network_channels)}")
+
+        # Check if the channels exist and belong to this network
+        invalid_channels = []
+        for channel in experiment_channels:
+            # Replace '_' with '-'. Eg: the_hague can be written this way in experiment.json while the comparison is
+            # done with the-hague defined in network/channels.json
+            node_slug1 = channel['node_slug1'].replace('_', '-')
```

Comment:
What are node_slug1 and node_slug2? Inmy experiment.json these are called node1 and node2